baremetal/metal3machine_manager.go

```diff
@@ -1856,25 +1859,32 @@ func (m *MachineManager) getMachineSet(ctx context.Context) (*clusterv1.MachineS
 	}
 
 	// Iterate over MachineSets list and find MachineSet which references specific machine.
+	var machineSetError string
 	for index := range machineSets.Items {
 		machineset := &machineSets.Items[index]
 		for _, mOwnerRef := range m.Machine.ObjectMeta.OwnerReferences {
 			if mOwnerRef.Kind != machineset.Kind {
 				continue
 			}
 			if mOwnerRef.APIVersion != machineset.APIVersion {
+				machineSetError += fmt.Sprintf("MachineSet %s has different API version %s than Machine %s with API version %s",
```

Comment:
These `machineSetError` is terminal I am guessing?

Reply:
There are no terminal errors in v1beta2, even if we set it.
Note: This change only improves the error message in the logs. Previously, we returned a generic errors.New("MachineSet is not found"). Now, the message provides more context about why it wasn't found. There is no change in functionality.